### PR TITLE
doc: improve/fix 'let' FAQ

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/doc/complement-design-faq.md [hunk 1]
```diff
@@ -165,12 +165,13 @@ particularly easy to read.
 
 ## Why is `let` used to introduce variables?
 
-We don't use the term "variable", instead, we use "variable bindings". The
-simplest way for binding is the `let` syntax, other ways including `if let`,
-`while let` and `match`. Bindings also exist in function arguments positions.
+Instead of the term "variable", we use "variable bindings". The
+simplest way for creating a binding is by using the `let` syntax.
+Other ways include `if let`, `while let`, and `match`. Bindings also
```
Comment (bombless) [inline src/doc/complement-design-faq.md]: One thing is missing here: for-in also do patten matching.

Comment (tshepang) [inline src/doc/complement-design-faq.md]: This has now be merged, so you can you create a separate PR.

Comment (bombless) [inline src/doc/complement-design-faq.md]: What do you think about "Other ways include `if let`, `while let`, `match` and `for ... in`. "?

Comment (tshepang) [inline src/doc/complement-design-faq.md]: While doing this PR, I was actually thinking maybe it should a bullet list:

> Other ways of crating bindings:
> - `if let`
> - `while let`
> ...

**Pull request merged**